src/QMCDrivers/VMC/VMCBatched.cpp

```diff
@@ -215,6 +215,216 @@ void VMCBatched::advanceWalkers(const StateForThread& sft,
+                                       const ParticleSet::Scalar_t& delta_spin) {
+                           return spindrift + (sqrtspintau * delta_spin);
+                         });
+        }
+        else
+        {
+          std::transform(delta_r_start, delta_r_end, drifts.begin(),
+                         [sqrttau](const PosType& delta_r) { return sqrttau * delta_r; });
+          std::transform(delta_spin_start, delta_spin_end, spindrifts.begin(),
+                         [sqrtspintau](const ParticleSet::Scalar_t& delta_spin) { return sqrtspintau * delta_spin; });
+        }
+
+        ps_dispatcher.flex_makeMoveWithSpin(walker_elecs, iat, drifts, spindrifts);
+
+        // This is inelegant
+        if (use_drift)
+        {
+          twf_dispatcher.flex_calcRatioGrad(walker_twfs, walker_elecs, iat, ratios, grads_new);
+          twf_dispatcher.flex_calcRatioGradWithSpin(walker_twfs, walker_elecs, iat, ratios, grads_new, spingrads_new);
+          std::transform(delta_r_start, delta_r_end, log_gf.begin(),
+                         [](const PosType& delta_r) { return mhalf * dot(delta_r, delta_r); });
+          std::transform(delta_spin_start, delta_spin_end, log_gf.begin(), log_gf.begin(),
+                         [](const ParticleSet::Scalar_t& delta_spin, const RealType& lgf) {
+                           return lgf + mhalf * delta_spin * delta_spin;
+                         });
+
+          sft.drift_modifier.getDrifts(tauovermass, grads_new, drifts);
+          sft.drift_modifier.getDrifts(spintauovermass, spingrads_new, spindrifts);
+
+          std::transform(crowd.beginElectrons(), crowd.endElectrons(), drifts.begin(), drifts.begin(),
+                         [iat](const ParticleSet& elecs, const PosType& drift) {
+                           return elecs.R[iat] - elecs.getActivePos() - drift;
+                         });
+          std::transform(crowd.beginElectrons(), crowd.endElectrons(), spindrifts.begin(), spindrifts.begin(),
+                         [iat](const ParticleSet& elecs, const ParticleSet::Scalar_t& spindrift) {
+                           return elecs.spins[iat] - elecs.getActiveSpinVal() - spindrift;
+                         });
+
+          std::transform(drifts.begin(), drifts.end(), log_gb.begin(),
+                         [oneover2tau](const PosType& drift) { return -oneover2tau * dot(drift, drift); });
+          std::transform(spindrifts.begin(), spindrifts.end(), log_gb.begin(), log_gb.begin(),
+                         [oneover2spintau](const ParticleSet::Scalar_t& spindrift, const RealType& lgb) {
+                           return lgb - oneover2spintau * spindrift * spindrift;
+                         });
+        }
+        else
+        {
+          twf_dispatcher.flex_calcRatio(walker_twfs, walker_elecs, iat, ratios);
+        }
+
+        std::transform(ratios.begin(), ratios.end(), prob.begin(), [](auto ratio) { return std::norm(ratio); });
```

Comment:
From this line and below, the code is spin agnostic. We need to think harder about an abstract to wrap the divergence between spin and non-spin cases.

Reply:
same is true in the DMC batched.